File: src/lib/types/ui/library/album/AlbumListView.ts
export interface AlbumListView {
  id: string;
  library_id: number;
  title: string;
  title_normalized: string;
  album_type: string;
  musicbrainz_id: string | null;
  artist_id: string | null;
  artist: string | null;
  year: number | null;
  genre: string | null;
  cover_url: string | null;
  total_tracks: number;
  total_duration: number;
  notes: string | null;
  /** L'artiste n'est qu'invité sur cet album. */
  participation: boolean;
  created_at: string;
  updated_at: string; 
}

File: src-tauri/src/mapper/library/album/album_list_view.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct AlbumListView {
    // ===== Identité Album =====
    pub id: String,
    pub library_id: i64,
    pub title: String,
    pub title_normalized: String,
    pub album_type: String, // album, single, ep, compilation
    pub musicbrainz_id: Option<String>,
    pub artist_id: Option<String>,
    pub artist: String, // nom uniquement pour listing
    pub year: Option<i32>,
    pub genre: Option<String>,
    pub cover_url: Option<String>,  // fallback éventuel depuis cache
    pub total_tracks: i64,
    pub total_duration: f64,
    pub notes: Option<String>,
    /// L'artiste n'est qu'invité sur cet album (compilation, BO, featuring).
    #[sqlx(default)]
    pub participation: bool,

    // ===== Timestamps =====
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
